agent/setup.go

```diff
@@ -316,14 +341,40 @@ func getPrometheusDefs(cfg lib.TelemetryConfig, isServer bool) ([]prometheus.Gau
 		raftCounters,
 		rate.Counters,
 	}
+
+	// For some unknown reason, we seem to add the raft counters above without
+	// checking if this is a server like we do above for some of the summaries
+	// above. We should probably fix that but I want to not change behavior right
+	// now. If we are a server, add summaries for WAL and verifier metrics.
+	if isServer && cfg.RaftLogStoreConfig.Verification.Enabled {
+		verifierCounters := make([]prometheus.CounterDefinition, 0)
+		for _, d := range verifier.MetricDefinitions.Counters {
+			verifierCounters = append(verifierCounters, prometheus.CounterDefinition{
+				Name: []string{"raft", "logstore", "verifier", d.Name},
+				Help: d.Desc,
+			})
+		}
+		counters = append(counters, verifierCounters)
+	}
+	if isServer && cfg.RaftLogStoreConfig.Backend == consul.LogStoreBackendWAL {
+		walCounters := make([]prometheus.CounterDefinition, 0)
```

Comment:
Might be a simple optimization to pass the cap here?
```suggestion
		walCounters := make([]prometheus.CounterDefinition, 0, len(wal.MetricDefinitions.Counters))
```